From 54895ab3cddd8596b9b5b321db5e0b7cee457382 Mon Sep 17 00:00:00 2001
From: "P.K. Lee" <pkleequanta@gmail.com>
Date: Thu, 26 Feb 2026 14:28:06 +0800
Subject: [PATCH] ARM: dts: aspeed: ventura: add the 0x11 ioexp to i2c10

1) Add a GPIO expander at address 0x11 on i2c10, and assign an SGPIO pin as
the IRQ for it.
2) Fix the "failed to match any schema with compatible" issues.
3) Reorder the nodes in alphabetically.

LORE Link:
https://lore.kernel.org/all/20260226055521.1655243-1-pkleequanta@gmail.com/T/#t

Tested:
Verified to function correctly on the Ventura platform.

Signed-off-by: P.K. Lee <pkleequanta@gmail.com>
---
 .../aspeed/aspeed-bmc-facebook-ventura.dts    | 444 +++++++++---------
 1 file changed, 229 insertions(+), 215 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
index ecf1420f2bd5..8570f121496c 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
@@ -201,7 +201,7 @@ led-22 {
 		led-23 {
 			label = "compute7_led_blue";
 			default-state = "off";
-			gpios = <&tray_leds_g4_gpio 3 GPIO_ACTIVE_LOW>; 
+			gpios = <&tray_leds_g4_gpio 3 GPIO_ACTIVE_LOW>;
 		};
 
 		led-24 {
@@ -659,50 +659,6 @@ tpm@0 {
 	};
 };
 
-&fmc {
-	status = "okay";
-	flash@0 {
-		status = "okay";
-		m25p,fast-read;
-		label = "bmc";
-		spi-max-frequency = <50000000>;
-#include "openbmc-flash-layout-128.dtsi"
-	};
-	flash@1 {
-		status = "okay";
-		m25p,fast-read;
-		label = "alt-bmc";
-		spi-max-frequency = <50000000>;
-	};
-};
-
-&mdio0 {
-	status = "okay";
-};
-
-&mac3 {
-	status = "okay";
-	phy-mode = "rmii";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_rmii4_default>;
-	fixed-link {
-		speed = <100>;
-		full-duplex;
-	};
-};
-
-&ehci0 {
-	status = "okay";
-};
-
-&ehci1 {
-	status = "okay";
-};
-
-&uhci {
-	status = "okay";
-};
-
 &adc0 {
 	vref-supply = <&p1v8_bmc_aux>;
 	status = "okay";
@@ -722,6 +678,31 @@ &adc1 {
 	pinctrl-0 = <&pinctrl_adc10_default>;
 };
 
+&ehci0 {
+	status = "okay";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&fmc {
+	status = "okay";
+	flash@0 {
+		status = "okay";
+		m25p,fast-read;
+		label = "bmc";
+		spi-max-frequency = <50000000>;
+#include "openbmc-flash-layout-128.dtsi"
+	};
+	flash@1 {
+		status = "okay";
+		m25p,fast-read;
+		label = "alt-bmc";
+		spi-max-frequency = <50000000>;
+	};
+};
+
 &gpio0 {
 	gpio-line-names =
 	/*A0-A7*/	"","","","","","","","",
@@ -752,170 +733,6 @@ &gpio0 {
 	/*Z0-Z7*/	"","","","","","","","";
 };
 
-&sgpiom0 {
-	status = "okay";
-	ngpios = <128>;
-	bus-frequency = <2000000>;
-
-	gpio-line-names =
-	/*"input pin","output pin"*/
-	/*A0 - A7*/
-	"power-chassis-good","power-chassis-control",
-	"host0-ready","WATER_VALVE_CLOSED_N",
-	"wPRSNT_RETURN_PLD_R_N","FM_MDIO_SW_SEL_PLD",
-	"wPRSNT_SUPPLY_PLD_R_N","FM_88E6393X_BIN_UPDATE_EN_N",
-	"LEAK3_DETECT","",
-	"LEAK4_DETECT","",
-	"RETURN_CNTL_FB_D_R","",
-	"SUPPLY_CNTL_FB_D_R","",
-	/*B0 - B7*/
-	"IOEXP0_INT_N","",
-	"IOEXP11_INT_N","",
-	"IOEXP10_INT_N","",
-	"IOEXP9_INT_N","",
-	"RPU_2_READY_SPARE_PLD_R","",
-	"RPU_2_READY_PLD_R","",
-	"RPU_READY_SPARE_PLD_R","",
-	"RPU_READY_PLD_R","",
-	/*C0 - C7*/
-	"wAALC_RPU_READY","",
-	"LEAK0_DETECT","",
-	"LEAK1_DETECT","",
-	"LEAK2_DETECT","",
-	"PRSNT_COMPUTE_TRAY1_N","",
-	"PRSNT_COMPUTE_TRAY2_N","",
-	"PRSNT_COMPUTE_TRAY3_N","",
-	"PRSNT_COMPUTE_TRAY4_N","",
-	/*D0 - D7*/
-	"PRSNT_COMPUTE_TRAY5_N","",
-	"PRSNT_COMPUTE_TRAY6_N","",
-	"PRSNT_COMPUTE_TRAY7_N","",
-	"PRSNT_COMPUTE_TRAY8_N","",
-	"PRSNT_NVS_TRAY1_N","",
-	"PRSNT_NVS_TRAY2_N","",
-	"PRSNT_COMPUTE_TRAY11_N","",
-	"PRSNT_COMPUTE_TRAY12_N","",
-	/*E0 - E7*/
-	"PRSNT_COMPUTE_TRAY13_N","",
-	"PRSNT_COMPUTE_TRAY14_N","",
-	"PRSNT_COMPUTE_TRAY15_N","",
-	"PRSNT_COMPUTE_TRAY16_N","",
-	"PRSNT_COMPUTE_TRAY17_N","",
-	"PRSNT_COMPUTE_TRAY18_N","",
-	"PRSNT_NVS_TRAY3_N","",
-	"PRSNT_NVS_TRAY4_N","",
-	/*F0 - F7*/
-	"PRSNT_NVS_TRAY5_N","",
-	"PRSNT_NVS_TRAY6_N","",
-	"PRSNT_NVS_TRAY7_N","",
-	"PRSNT_NVS_TRAY8_N","",
-	"PRSNT_NVS_TRAY9_N","",
-	"PRSNT_COMPUTE_TRAY9_N","",
-	"PRSNT_COMPUTE_TRAY10_N","",
-	"SMALL_LEAK_COMPUTE_TRAY1_N","",
-	/*G0 - G7*/
-	"SMALL_LEAK_COMPUTE_TRAY2_N","",
-	"SMALL_LEAK_COMPUTE_TRAY3_N","",
-	"SMALL_LEAK_COMPUTE_TRAY4_N","",
-	"SMALL_LEAK_COMPUTE_TRAY5_N","",
-	"SMALL_LEAK_COMPUTE_TRAY6_N","",
-	"SMALL_LEAK_COMPUTE_TRAY7_N","",
-	"SMALL_LEAK_COMPUTE_TRAY8_N","",
-	"SMALL_LEAK_NVS_TRAY1_N","",
-	/*H0 - H7*/
-	"SMALL_LEAK_NVS_TRAY2_N","",
-	"SMALL_LEAK_COMPUTE_TRAY11_N","",
-	"SMALL_LEAK_COMPUTE_TRAY12_N","",
-	"SMALL_LEAK_COMPUTE_TRAY13_N","",
-	"SMALL_LEAK_COMPUTE_TRAY14_N","",
-	"SMALL_LEAK_COMPUTE_TRAY15_N","",
-	"SMALL_LEAK_COMPUTE_TRAY16_N","",
-	"SMALL_LEAK_COMPUTE_TRAY17_N","",
-	/*I0 - I7*/
-	"SMALL_LEAK_COMPUTE_TRAY18_N","",
-	"SMALL_LEAK_NVS_TRAY3_N","",
-	"SMALL_LEAK_NVS_TRAY4_N","",
-	"SMALL_LEAK_NVS_TRAY5_N","",
-	"SMALL_LEAK_NVS_TRAY6_N","",
-	"SMALL_LEAK_NVS_TRAY7_N","",
-	"SMALL_LEAK_NVS_TRAY8_N","",
-	"SMALL_LEAK_NVS_TRAY9_N","",
-	/*J0 - J7*/
-	"SMALL_LEAK_COMPUTE_TRAY9_N","",
-	"SMALL_LEAK_COMPUTE_TRAY10_N","",
-	"PWRGD_COMPUTE_TRAY1_N","",
-	"PWRGD_COMPUTE_TRAY2_N","",
-	"PWRGD_COMPUTE_TRAY3_N","",
-	"PWRGD_COMPUTE_TRAY4_N","",
-	"PWRGD_COMPUTE_TRAY5_N","",
-	"PWRGD_COMPUTE_TRAY6_N","",
-	/*K0 - K7*/
-	"PWRGD_COMPUTE_TRAY7_N","",
-	"PWRGD_COMPUTE_TRAY8_N","",
-	"PWRGD_NVS_TRAY1_PWROK_N","",
-	"PWRGD_NVS_TRAY2_PWROK_N","",
-	"PWRGD_COMPUTE_TRAY11_N","",
-	"PWRGD_COMPUTE_TRAY12_N","",
-	"PWRGD_COMPUTE_TRAY13_N","",
-	"PWRGD_COMPUTE_TRAY14_N","",
-	/*L0 - L7*/
-	"PWRGD_COMPUTE_TRAY15_N","",
-	"PWRGD_COMPUTE_TRAY16_N","",
-	"PWRGD_COMPUTE_TRAY17_N","",
-	"PWRGD_COMPUTE_TRAY18_N","",
-	"PWRGD_NVS_TRAY3_PWROK_N","",
-	"PWRGD_NVS_TRAY4_PWROK_N","",
-	"PWRGD_NVS_TRAY5_PWROK_N","",
-	"PWRGD_NVS_TRAY6_PWROK_N","",
-	/*M0 - M7*/
-	"PWRGD_NVS_TRAY7_PWROK_N","",
-	"PWRGD_NVS_TRAY8_PWROK_N","",
-	"PWRGD_NVS_TRAY9_PWROK_N","",
-	"PWRGD_COMPUTE_TRAY9_N","",
-	"PWRGD_COMPUTE_TRAY10_N","",
-	"LEAK_DETECT_COMPUTE_TRAY1_N","",
-	"LEAK_DETECT_COMPUTE_TRAY2_N","",
-	"LEAK_DETECT_COMPUTE_TRAY3_N","",
-	/*N0 - N7*/
-	"LEAK_DETECT_COMPUTE_TRAY4_N","",
-	"LEAK_DETECT_COMPUTE_TRAY5_N","",
-	"LEAK_DETECT_COMPUTE_TRAY6_N","",
-	"LEAK_DETECT_COMPUTE_TRAY7_N","",
-	"LEAK_DETECT_COMPUTE_TRAY8_N","",
-	"LEAK_DETECT_NVS_TRAY1_N","",
-	"LEAK_DETECT_NVS_TRAY2_N","",
-	"LEAK_DETECT_COMPUTE_TRAY11_N","",
-	/*O0 - O7*/
-	"LEAK_DETECT_COMPUTE_TRAY12_N","",
-	"LEAK_DETECT_COMPUTE_TRAY13_N","",
-	"LEAK_DETECT_COMPUTE_TRAY14_N","",
-	"LEAK_DETECT_COMPUTE_TRAY15_N","",
-	"LEAK_DETECT_COMPUTE_TRAY16_N","",
-	"LEAK_DETECT_COMPUTE_TRAY17_N","",
-	"LEAK_DETECT_COMPUTE_TRAY18_N","",
-	"LEAK_DETECT_NVS_TRAY3_N","",
-	/*P0 - P7*/
-	"LEAK_DETECT_NVS_TRAY4_N","",
-	"LEAK_DETECT_NVS_TRAY5_N","",
-	"LEAK_DETECT_NVS_TRAY6_N","",
-	"LEAK_DETECT_NVS_TRAY7_N","",
-	"LEAK_DETECT_NVS_TRAY8_N","",
-	"LEAK_DETECT_NVS_TRAY9_N","",
-	"LEAK_DETECT_COMPUTE_TRAY9_N","",
-	"LEAK_DETECT_COMPUTE_TRAY10_N","";
-};
-
-&wdt1 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_wdtrst1_default>;
-	aspeed,reset-type = "soc";
-	aspeed,external-signal;
-	aspeed,ext-push-pull;
-	aspeed,ext-active-high;
-	aspeed,ext-pulse-duration = <256>;
-};
-
 &i2c0 {
 	status = "okay";
 
@@ -1233,7 +1050,7 @@ i2c3mux0ch5: i2c@5 {
 			reg = <5>;
 
 			pwm@20 {
-				compatible = "max31790";
+				compatible = "maxim,max31790";
 				reg = <0x20>;
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -1442,6 +1259,25 @@ gpio@10 {
 		"IT_STOP_PUMP_SPARE", "IT_STOP_PUMP",
 		"", "",
 		"", "",
+		"wPWRGD_P12V_SCM_R", "wPWRGD_P12V_AUX_R",
+		"wPWRGD_P24V_AUX_R", "wPWRGD_P52V_HSC_PWROK_R",
+		"wSMB_TMC75_TEMP_ALERT_N_R", "wP48V_HSC_ALERT_N",
+		"wP24V_AUX_INA230_ALERT_N_R", "wP24V_SM_INA230_ALERT_N_R";
+	};
+
+	gpio@11 {
+		compatible = "nxp,pca9555";
+		reg = <0x11>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-parent = <&sgpiom0>;
+		interrupts = <2 IRQ_TYPE_LEVEL_LOW>;
+
+		gpio-line-names =
+		"", "",
+		"", "wPWRGD_P24V_SMPWROK",
+		"wPWRGD_P1V05_AUX_R", "wPWRGD_P1V5_AUX_R",
+		"wPWRGD_P3V3_AUX_R", "wPWRGD_P5V_AUX_R",
 		"", "",
 		"", "",
 		"", "",
@@ -1514,11 +1350,6 @@ power-monitor@44 {
 		compatible = "lltc,ltc4286";
 		reg = <0x44>;
 	};
-
-	power-monitor@69 {
-		compatible = "pmbus";
-		reg = <0x69>;
-	};
 };
 
 &i2c14 {
@@ -1576,3 +1407,186 @@ eeprom@56 {
 		reg = <0x56>;
 	};
 };
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rmii";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii4_default>;
+	fixed-link {
+		speed = <100>;
+		full-duplex;
+	};
+};
+
+&mdio0 {
+	status = "okay";
+};
+
+&sgpiom0 {
+	status = "okay";
+	ngpios = <128>;
+	bus-frequency = <2000000>;
+
+	gpio-line-names =
+	/*"input pin","output pin"*/
+	/*A0 - A7*/
+	"power-chassis-good","power-chassis-control",
+	"IOEXP1_INT_N","WATER_VALVE_CLOSED_N",
+	"wPRSNT_RETURN_PLD_R_N","FM_MDIO_SW_SEL_PLD",
+	"wPRSNT_SUPPLY_PLD_R_N","FM_88E6393X_BIN_UPDATE_EN_N",
+	"LEAK3_DETECT","",
+	"LEAK4_DETECT","",
+	"RETURN_CNTL_FB_D_R","",
+	"SUPPLY_CNTL_FB_D_R","",
+	/*B0 - B7*/
+	"IOEXP0_INT_N","",
+	"IOEXP11_INT_N","",
+	"IOEXP10_INT_N","",
+	"IOEXP9_INT_N","",
+	"RPU_2_READY_SPARE_PLD_R","",
+	"RPU_2_READY_PLD_R","",
+	"RPU_READY_SPARE_PLD_R","",
+	"RPU_READY_PLD_R","",
+	/*C0 - C7*/
+	"wAALC_RPU_READY","",
+	"LEAK0_DETECT","",
+	"LEAK1_DETECT","",
+	"LEAK2_DETECT","",
+	"PRSNT_COMPUTE_TRAY1_N","",
+	"PRSNT_COMPUTE_TRAY2_N","",
+	"PRSNT_COMPUTE_TRAY3_N","",
+	"PRSNT_COMPUTE_TRAY4_N","",
+	/*D0 - D7*/
+	"PRSNT_COMPUTE_TRAY5_N","",
+	"PRSNT_COMPUTE_TRAY6_N","",
+	"PRSNT_COMPUTE_TRAY7_N","",
+	"PRSNT_COMPUTE_TRAY8_N","",
+	"PRSNT_NVS_TRAY1_N","",
+	"PRSNT_NVS_TRAY2_N","",
+	"PRSNT_COMPUTE_TRAY11_N","",
+	"PRSNT_COMPUTE_TRAY12_N","",
+	/*E0 - E7*/
+	"PRSNT_COMPUTE_TRAY13_N","",
+	"PRSNT_COMPUTE_TRAY14_N","",
+	"PRSNT_COMPUTE_TRAY15_N","",
+	"PRSNT_COMPUTE_TRAY16_N","",
+	"PRSNT_COMPUTE_TRAY17_N","",
+	"PRSNT_COMPUTE_TRAY18_N","",
+	"PRSNT_NVS_TRAY3_N","",
+	"PRSNT_NVS_TRAY4_N","",
+	/*F0 - F7*/
+	"PRSNT_NVS_TRAY5_N","",
+	"PRSNT_NVS_TRAY6_N","",
+	"PRSNT_NVS_TRAY7_N","",
+	"PRSNT_NVS_TRAY8_N","",
+	"PRSNT_NVS_TRAY9_N","",
+	"PRSNT_COMPUTE_TRAY9_N","",
+	"PRSNT_COMPUTE_TRAY10_N","",
+	"SMALL_LEAK_COMPUTE_TRAY1_N","",
+	/*G0 - G7*/
+	"SMALL_LEAK_COMPUTE_TRAY2_N","",
+	"SMALL_LEAK_COMPUTE_TRAY3_N","",
+	"SMALL_LEAK_COMPUTE_TRAY4_N","",
+	"SMALL_LEAK_COMPUTE_TRAY5_N","",
+	"SMALL_LEAK_COMPUTE_TRAY6_N","",
+	"SMALL_LEAK_COMPUTE_TRAY7_N","",
+	"SMALL_LEAK_COMPUTE_TRAY8_N","",
+	"SMALL_LEAK_NVS_TRAY1_N","",
+	/*H0 - H7*/
+	"SMALL_LEAK_NVS_TRAY2_N","",
+	"SMALL_LEAK_COMPUTE_TRAY11_N","",
+	"SMALL_LEAK_COMPUTE_TRAY12_N","",
+	"SMALL_LEAK_COMPUTE_TRAY13_N","",
+	"SMALL_LEAK_COMPUTE_TRAY14_N","",
+	"SMALL_LEAK_COMPUTE_TRAY15_N","",
+	"SMALL_LEAK_COMPUTE_TRAY16_N","",
+	"SMALL_LEAK_COMPUTE_TRAY17_N","",
+	/*I0 - I7*/
+	"SMALL_LEAK_COMPUTE_TRAY18_N","",
+	"SMALL_LEAK_NVS_TRAY3_N","",
+	"SMALL_LEAK_NVS_TRAY4_N","",
+	"SMALL_LEAK_NVS_TRAY5_N","",
+	"SMALL_LEAK_NVS_TRAY6_N","",
+	"SMALL_LEAK_NVS_TRAY7_N","",
+	"SMALL_LEAK_NVS_TRAY8_N","",
+	"SMALL_LEAK_NVS_TRAY9_N","",
+	/*J0 - J7*/
+	"SMALL_LEAK_COMPUTE_TRAY9_N","",
+	"SMALL_LEAK_COMPUTE_TRAY10_N","",
+	"PWRGD_COMPUTE_TRAY1_N","",
+	"PWRGD_COMPUTE_TRAY2_N","",
+	"PWRGD_COMPUTE_TRAY3_N","",
+	"PWRGD_COMPUTE_TRAY4_N","",
+	"PWRGD_COMPUTE_TRAY5_N","",
+	"PWRGD_COMPUTE_TRAY6_N","",
+	/*K0 - K7*/
+	"PWRGD_COMPUTE_TRAY7_N","",
+	"PWRGD_COMPUTE_TRAY8_N","",
+	"PWRGD_NVS_TRAY1_PWROK_N","",
+	"PWRGD_NVS_TRAY2_PWROK_N","",
+	"PWRGD_COMPUTE_TRAY11_N","",
+	"PWRGD_COMPUTE_TRAY12_N","",
+	"PWRGD_COMPUTE_TRAY13_N","",
+	"PWRGD_COMPUTE_TRAY14_N","",
+	/*L0 - L7*/
+	"PWRGD_COMPUTE_TRAY15_N","",
+	"PWRGD_COMPUTE_TRAY16_N","",
+	"PWRGD_COMPUTE_TRAY17_N","",
+	"PWRGD_COMPUTE_TRAY18_N","",
+	"PWRGD_NVS_TRAY3_PWROK_N","",
+	"PWRGD_NVS_TRAY4_PWROK_N","",
+	"PWRGD_NVS_TRAY5_PWROK_N","",
+	"PWRGD_NVS_TRAY6_PWROK_N","",
+	/*M0 - M7*/
+	"PWRGD_NVS_TRAY7_PWROK_N","",
+	"PWRGD_NVS_TRAY8_PWROK_N","",
+	"PWRGD_NVS_TRAY9_PWROK_N","",
+	"PWRGD_COMPUTE_TRAY9_N","",
+	"PWRGD_COMPUTE_TRAY10_N","",
+	"LEAK_DETECT_COMPUTE_TRAY1_N","",
+	"LEAK_DETECT_COMPUTE_TRAY2_N","",
+	"LEAK_DETECT_COMPUTE_TRAY3_N","",
+	/*N0 - N7*/
+	"LEAK_DETECT_COMPUTE_TRAY4_N","",
+	"LEAK_DETECT_COMPUTE_TRAY5_N","",
+	"LEAK_DETECT_COMPUTE_TRAY6_N","",
+	"LEAK_DETECT_COMPUTE_TRAY7_N","",
+	"LEAK_DETECT_COMPUTE_TRAY8_N","",
+	"LEAK_DETECT_NVS_TRAY1_N","",
+	"LEAK_DETECT_NVS_TRAY2_N","",
+	"LEAK_DETECT_COMPUTE_TRAY11_N","",
+	/*O0 - O7*/
+	"LEAK_DETECT_COMPUTE_TRAY12_N","",
+	"LEAK_DETECT_COMPUTE_TRAY13_N","",
+	"LEAK_DETECT_COMPUTE_TRAY14_N","",
+	"LEAK_DETECT_COMPUTE_TRAY15_N","",
+	"LEAK_DETECT_COMPUTE_TRAY16_N","",
+	"LEAK_DETECT_COMPUTE_TRAY17_N","",
+	"LEAK_DETECT_COMPUTE_TRAY18_N","",
+	"LEAK_DETECT_NVS_TRAY3_N","",
+	/*P0 - P7*/
+	"LEAK_DETECT_NVS_TRAY4_N","",
+	"LEAK_DETECT_NVS_TRAY5_N","",
+	"LEAK_DETECT_NVS_TRAY6_N","",
+	"LEAK_DETECT_NVS_TRAY7_N","",
+	"LEAK_DETECT_NVS_TRAY8_N","",
+	"LEAK_DETECT_NVS_TRAY9_N","",
+	"LEAK_DETECT_COMPUTE_TRAY9_N","",
+	"LEAK_DETECT_COMPUTE_TRAY10_N","";
+};
+
+&uhci {
+	status = "okay";
+};
+
+&wdt1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdtrst1_default>;
+	aspeed,reset-type = "soc";
+	aspeed,external-signal;
+	aspeed,ext-push-pull;
+	aspeed,ext-active-high;
+	aspeed,ext-pulse-duration = <256>;
+};
-- 
2.53.0-Meta

